app/src/main/java/com/hedvig/app/feature/marketing/ui/MarketingActivity.kt

```diff
@@ -4,96 +4,92 @@ import android.content.Context
-                Market.NO -> {
-                    startActivity(WebOnboardingActivity.newInstance(this))
+        model.navigationState.observe(this) { navigationState ->
+            when (navigationState.destination) {
+                MARKET_PICKER -> {
+                    val transaction = supportFragmentManager.beginTransaction()
+                    navigationState.sharedElements.forEach { (view, transitionName) ->
+                        transaction.addSharedElement(view, transitionName)
+                    }
+                    transaction.replace(R.id.container, MarketPickerFragment().also {
+                        it.sharedElementEnterTransition = MaterialContainerTransform()
+                    }).commit()
                 }
-            }
-        }
-
-        logIn.setHapticClickListener {
-            tracker.logIn()
-            when (market) {
-                Market.SE -> {
-                    AuthenticateDialog().show(supportFragmentManager, AuthenticateDialog.TAG)
-                }
-                Market.NO -> {
-                    startActivity(NorwegianAuthenticationActivity.newInstance(this))
+                MARKETING -> {
+                    val transaction = supportFragmentManager.beginTransaction()
+                    navigationState.sharedElements.forEach { (view, transitionName) ->
+                        transaction.addSharedElement(view, transitionName)
+                    }
+                    transaction.replace(R.id.container, MarketSelectedFragment().also {
+                        it.sharedElementEnterTransition = MaterialContainerTransform()
+                    }).commit()
                 }
             }
         }
 
-        model
-            .marketingBackground
-            .observe(this) { image ->
-                image?.let { i ->
-                    val placeholder = BlurHashDecoder.decode(i.blurhash, 32, 32)
+        binding.apply {
+            root.setEdgeToEdgeSystemUiFlags(true)
 
+            model
+                .marketingBackground
+                .observe(this@MarketingActivity) { image ->
+                    val placeholder = BlurHashDecoder.decode(image.blurhash, 32, 32)
                     Glide
-                        .with(this)
-                        .load(i.image?.url)
+                        .with(this@MarketingActivity)
+                        .load(image.image?.url)
                         .transition(withCrossFade())
                         .placeholder(BitmapDrawable(resources, placeholder))
                         .into(backgroundImage)
+
+                    when (image.userInterfaceStyle) {
+                        UserInterfaceStyle.LIGHT -> {
+                            val view = window.decorView
+                            view.systemUiVisibility =
+                                view.systemUiVisibility and View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR.inv()
+                        }
+                        UserInterfaceStyle.DARK -> {
```

Comment:
```suggestion
                        UserInterfaceStyle.DARK, UserInterfaceStyle.UNKNOWN__ -> {
```

Then you no longer need to have two cases saying the same thing 😉